node-notes-cli/app.js

```diff
@@ -0,0 +1,36 @@
+const fs = require('fs');
+const json = require('./data.json');
+const action = process.argv[2];
+
+if (action === 'read') {
+  for (let i = 1; i < `${json.nextId}`; i++) {
+    if (`${json.notes[i]}` !== 'undefined') {
+      console.log(`${i}: ${json.notes[i]}`);
+    }
+  }
+} else if (action === 'add') {
+  const note = process.argv[3];
+  json.notes[json.nextId++] = note;
+  fs.writeFile('data.json', `${JSON.stringify(json, null, 2)}`, err => {
+    if (err) throw err;
+  }
+  );
+} else if (action === 'update') {
+  const update = process.argv[3];
+  for (let i = 1; i < `${json.nextId}`; i++) {
+    if (`${json.notes[i]}` === json.notes[update]) {
+      json.notes[i] = process.argv[4];
```

Comment:
It only updates existing notes now, but it can also update more than one note if they happen to share the same note text. Since the user specifies a note id to update, the script should only update that particular note (if it exists), not any other notes that happen to match it.